source/mongodb-handshake/handshake.rst

```diff
@@ -52,22 +53,43 @@ Specification
 Connection handshake
 --------------------
 
-The ``isMaster`` handshake MUST be performed on every socket to any and all servers
+MongoDB uses the ``hello`` or ``isMaster`` commands for handshakes and topology
+monitoring. ``hello`` is the modern and preferred command. ``isMaster`` is referred
+to as "legacy hello" and is maintained for backwards compatibility with servers
+that do not support the ``hello`` command.
+
+NOTE: The legacy hello command name is ``isMaster``, but servers will accept the
+alternate casing ``ismaster``. Other case variations result in ``CommandNotFound``.
+Drivers MUST take this case variation into account when determining which commands
+to encrypt, redact, or otherwise treat specially.
+
+If Versioned API is requested, drivers MUST use the ``hello`` command for the
+initial handshake. If Versioned API is not requested, drivers MUST use legacy
+hello (``isMaster``) for the initial handshake and include ``helloOk: true``
+in the handshake request.
+
+ASIDE: If the legacy handshake response includes ``helloOk: true``, then
```

Comment:
This seems less like an aside since this has a "MUST" statement.

```suggestion
If the legacy handshake response includes ``helloOk: true``, then
```

Reply:
The reason I made it an aside is because this is really an SDAM issue - and will be dealt with in detail in the SDAM spec - but felt it should be mentioned in the handshake spec for context on why `helloOk:true` is necessary.

Reply:
That makes sense!